lib/twilio/wstransport.ts

```diff
@@ -515,6 +537,34 @@ export default class WSTransport extends EventEmitter {
     this.state = state;
   }
 
+  /**
+   * Perform a backoff based on the signaling retryAfter value.
+   */
+  private _retryAfterBackoff(retryAfter: number): void {
+    if (this.state === WSTransportState.Closed) {
+      this._log.info('RetryAfter backoff initiated but transport state is closed; not attempting a connection.');
+      return;
+    }
+    if (!this._backoffStartTime.retryAfter) {
+      this._log.info('Initializing retryAfter backoff');
+      this._backoffStartTime.retryAfter = Date.now();
+      this._log.info(`RetryAfter backoff start; ${this._backoffStartTime.retryAfter}`);
+    }
+    if (Date.now() - this._backoffStartTime.retryAfter > this._options.maxRetryAfterDurationMs) {
+      this._log.info('Max retryAfter backoff attempt time exceeded; not attempting a connection.');
+      return;
+    }
+
+    this._log.info(`Received retryAfter from signaling server: ${retryAfter}(s)`);
+    if (this._preferredUri) {
+      this._log.info('Preferred URI set; backing off using retryAfter.');
+      this._connect(this._preferredUri, this._retryAfterAttempt++);
+    } else {
+      this._log.info('Preferred URI not set; backing off using retryAfter.');
+      this._connect(this._uris[this._uriIndex], this._retryAfterAttempt++);
```

Comment:
I see that you're calling _retryAfterBackoff as soon as you see it from signaling, which in turn will call this _connect method. However, the websocket is still connected once you get to this point of the code because this is called synchronously right? your current flow is
1. received retryAfter from signaling
2. invoked _connect
3. server disconnects websocket

As you can see, it doesn't make sense to connect before websocket disconnects